core/src/client/mod.rs

```diff
@@ -675,4 +698,516 @@ impl StarknetClient for StarknetClientImpl {
 
         Ok(Address::from(evm_address_sliced))
     }
+
+    async fn starknet_tx_into_eth_tx(
+        &self,
+        tx: StarknetTransaction,
+        block_hash: Option<PrimitiveH256>,
+        block_number: Option<U256>,
+    ) -> Result<EtherTransaction, KakarotClientError> {
+        let mut ether_tx = EtherTransaction::default();
+        println!("2.1 Inside Getting transactions");
+
+        match tx {
+            StarknetTransaction::Invoke(invoke_tx) => {
+                match invoke_tx {
+                    InvokeTransaction::V0(v0) => {
+                        println!("2.X Inside InvokeV0");
+
+                        // Extract relevant fields from InvokeTransactionV0 and convert them to the corresponding fields in EtherTransaction
+                        ether_tx.hash =
+                            PrimitiveH256::from_slice(&v0.transaction_hash.to_bytes_be());
+                        let tx_receipt = self
+                            .get_transaction_receipt(H256::from_slice(
```

Comment:
why does our custom client take an H256 as param for transaction receipt? sadge, didn't catch that earlier
nvm will fix it in another PR